src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -69,23 +99,19 @@ - (void)runLoop
         unsigned int delaySeconds = (10 * 60); // 10 minutes.
         DDLogDebug(
             @"%@ No more expiring messages. Setting next check %u seconds into the future", self.tag, delaySeconds);
-        [self runBy:now + delaySeconds * 1000];
+        [self runByDate:[NSDate ows_dateWithMillisecondsSince1970:now + delaySeconds * 1000]];
         return;
     }
 
     uint64_t nextExpirationAt = [nextExpirationTimestampNumber unsignedLongLongValue];
-    uint64_t runByMilliseconds;
-    if (nextExpirationAt < now + 1000) {
-        DDLogWarn(@"%@ Next run requested at %llu, which is too soon. Delaying by 1 sec to prevent churn",
-            self.tag,
-            nextExpirationAt);
-        runByMilliseconds = now + 1000;
-    } else {
-        runByMilliseconds = nextExpirationAt;
-    }
+    [self runByDate:[NSDate ows_dateWithMillisecondsSince1970:MAX(nextExpirationAt, now)]];
+}
 
-    DDLogVerbose(@"%@ Requesting next expiration to run by: %llu", self.tag, nextExpirationAt);
-    [self runBy:runByMilliseconds];
++ (void)setExpirationForMessage:(TSMessage *)message
+{
+    dispatch_async(self.serialQueue, ^{
```

Comment:
These public class method make it easier to use the singleton and enforce the invariant that all work done by this class is done on its private queue.